tfhe/src/core_crypto/algorithms/glwe_linear_algebra.rs

```diff
@@ -0,0 +1,1045 @@
+/// // computations
+/// // Define parameters for GlweCiphertext creation
+/// let glwe_dimension = GlweDimension(1);
+/// let polynomial_size = PolynomialSize(2048);
+/// let glwe_modular_std_dev = StandardDev(0.00000000000000029403601535432533);
+/// let ciphertext_modulus = CiphertextModulus::new_native();
+///
+/// // Create the PRNG
+/// let mut seeder = new_seeder();
+/// let seeder = seeder.as_mut();
+/// let mut encryption_generator =
+///     EncryptionRandomGenerator::<ActivatedRandomGenerator>::new(seeder.seed(), seeder);
+/// let mut secret_generator =
+///     SecretRandomGenerator::<ActivatedRandomGenerator>::new(seeder.seed());
+///
+/// // Create the GlweSecretKey
+/// let glwe_secret_key = allocate_and_generate_new_binary_glwe_secret_key(
+///     glwe_dimension,
+///     polynomial_size,
+///     &mut secret_generator,
+/// );
+///
+/// // Create the plaintext
+/// let msg = 3u64;
+/// let encoded_msg = msg << 60;
+/// let plaintext_list = PlaintextList::new(encoded_msg, PlaintextCount(polynomial_size.0));
+///
+/// // Create a new GlweCiphertext
+/// let mut glwe = GlweCiphertext::new(
+///     0u64,
+///     glwe_dimension.to_glwe_size(),
+///     polynomial_size,
+///     ciphertext_modulus,
+/// );
+/// encrypt_glwe_ciphertext(
+///     &glwe_secret_key,
+///     &mut glwe,
+///     &plaintext_list,
+///     glwe_modular_std_dev,
+///     &mut encryption_generator,
+/// );
+///
+/// glwe_ciphertext_plaintext_list_sub_assign(&mut glwe, &plaintext_list);
+///
+/// let mut output_plaintext_list =
+///     PlaintextList::new(0u64, PlaintextCount(glwe.polynomial_size().0));
+///
+/// decrypt_glwe_ciphertext(&glwe_secret_key, &glwe, &mut output_plaintext_list);
+///
+/// // Round and remove encoding
+/// // First create a decomposer working on the high 4 bits corresponding to our encoding.
+/// let decomposer = SignedDecomposer::new(DecompositionBaseLog(4), DecompositionLevelCount(1));
+///
+/// // Round and remove encoding in the output plaintext list
+/// output_plaintext_list
+///     .iter_mut()
+///     .for_each(|x| *x.0 = decomposer.closest_representable(*x.0) >> 60);
+///
+/// // Check we recovered the expected result
+/// assert!(output_plaintext_list.iter().all(|x| *x.0 == msg - msg));
+/// ```
+pub fn glwe_ciphertext_plaintext_list_sub_assign<Scalar, InCont, PlainCont>(
+    lhs: &mut GlweCiphertext<InCont>,
+    rhs: &PlaintextList<PlainCont>,
+) where
+    Scalar: UnsignedInteger,
+    InCont: ContainerMut<Element = Scalar>,
+    PlainCont: Container<Element = Scalar>,
+{
+    let mut body = lhs.get_mut_body();
+    let ciphertext_modulus = body.ciphertext_modulus();
+    assert!(ciphertext_modulus.is_compatible_with_native_modulus());
+
+    if ciphertext_modulus.is_native_modulus() {
+        body.as_mut()
+            .iter_mut()
+            .zip(rhs.iter())
+            .for_each(|(dst_body, src_plain)| *dst_body = (*dst_body).wrapping_sub(*src_plain.0));
```

Comment:
Same idea, `slice_wrapping_sub_assign(body.as_mut(), rhs.as_ref()) be possible ?

Reply:
definitely